File: api/src/main/java/com/ruby/rubia_server/core/entity/CampaignContact.java
package com.ruby.rubia_server.core.entity;

import com.ruby.rubia_server.core.enums.CampaignContactStatus;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;
import java.util.UUID;

import lombok.Getter;
import lombok.Setter;

@Entity
@Table(name = "campaign_contacts")
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class CampaignContact {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "campaign_id", nullable = false)
    private Campaign campaign;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "customer_id", nullable = false) // Era 'contact_id'
    private Customer customer; // O cliente (agora a entidade consolidada) associado a esta campanha

    @Enumerated(EnumType.STRING)
    @Column(name = "contact_status", nullable = false)
    @Builder.Default
    private CampaignContactStatus status = CampaignContactStatus.PENDING;

    @Column(name = "message_sent_at")
    private LocalDateTime messageSentAt;

    @Column(name = "response_received_at")
    private LocalDateTime responseReceivedAt;

    @Column(columnDefinition = "TEXT")
    private String notes;

    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
}

File: api-dotnet/Rubia.Server/Entities/CampaignContact.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Rubia.Server.Enums;

namespace Rubia.Server.Entities;

[Table("campaign_contacts")]
public class CampaignContact : BaseEntity
{
    [Column("contact_status")]
    [Required]
    public CampaignContactStatus Status { get; set; } = CampaignContactStatus.Pending;

    [Column("message_sent_at")]
    public DateTime? MessageSentAt { get; set; }

    [Column("response_received_at")]
    public DateTime? ResponseReceivedAt { get; set; }

    [Column("notes", TypeName = "TEXT")]
    public string? Notes { get; set; }

    // Navigation properties
    [Column("campaign_id")]
    [Required]
    public Guid CampaignId { get; set; }
    
    [ForeignKey("CampaignId")]
    public virtual Campaign Campaign { get; set; } = null!;

    [Column("customer_id")]
    [Required]
    public Guid CustomerId { get; set; }
    
    [ForeignKey("CustomerId")]
    public virtual Customer Customer { get; set; } = null!; // O cliente associado a esta campanha

    public virtual ICollection<Message> Messages { get; set; } = new List<Message>();
}